File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        String mergedString = "";
        int mergeLimit = Math.min(word1.length(), word2.length());
        for(int i = 0; i < mergeLimit; i++) {
            mergedString += word1.charAt(i);
            mergedString += word2.charAt(i);
        }
        if(word1.length() >= mergeLimit) {
            mergedString += word1.substring(mergeLimit);
        }
        if(word2.length() >= mergeLimit) {
            mergedString += word2.substring(mergeLimit);
        }
        return mergedString;
    }
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        mergedString = ''
        m = len(word1)
        n = len(word2)
        i,j = 0,0
        while i < m or j < n:
            if i < m:
                mergedString += word1[i]
                i += 1
            if j < n:
                mergedString += word2[j]
                j += 1
        return mergedString